Switch consumer to per-signal package

dmathieu opened the pull request:
<!--Ex. Fixing a bug - Describe the bug and how this fixes the issue.
Ex. Adding a feature - Explain what this achieves.-->
#### Description

This is a proposal showing the changes in the `consumer` package, as described in #10207.
With this change, the consumer package (and subpackages) can become stable, and new signals (such as profiles) can be added as new modules that aren't stable.

<!--Describe what testing was performed and which tests were added.-->
#### Testing

This is unit-tested.

bogdandrutu reviewed (commented):
Feedback about implementation (does not mean I agree with the split decision):
1. `clog` - is a bad name since we may have to use something like this for component as well and may confuse between consumer vs component, maybe `conslog`?
2. Do we also need to split `consumertest` package?  Do we need to rename it to be consistent to something like `ctest` (or `constest`)?
3. What about `consumererror`? Same question about naming.

dmathieu commented:
I have renamed `clog` to `conslog` (did you want the same rename for `ctrace` and `cmetric`?)
I have also split `consumertest` into per-signal packages. I'm looking into `consumererror`.
